hazelcast/src/main/java/com/hazelcast/cache/impl/CacheProxyUtil.java

```diff
@@ -236,4 +239,25 @@ public static int getPartitionId(NodeEngine nodeEngine, Data key) {
             }
         }
     }
+
+    public static <K, V> void validateCacheConfig(CacheConfig<K, V> cacheConfig) {
```

Comment:
Actually one works on `CacheSimpleConfig` and the other on `CacheConfig`. I didn't want to introduce any jcache hierarchy classes in ConfigValidator (in config package) so `checkCacheConfig` works on `CacheSimpleConfig` ( no jcache dependencies) while `validateCacheConfig` implements the same validations on `CacheConfig`.